geppetto.js/geppetto-ui/src/3d-canvas/showcase/examples/AuditoryCortexExample.js

```diff
@@ -0,0 +1,228 @@
+import Button from "@material-ui/core/Button";
+
+const INSTANCE_NAME = 'acnet2';
+const COLORS = [
+  { r: 0, g: 0.2, b: 0.6, a: 1 },
+  { r: 0.8, g: 0, b: 0, a: 1 },
+  { r: 0, g: 0.8, b: 0, a: 1 },
+  { r: 0, g: 0.8, b: 0, a: 0.5 },
+];
+const SELECTION_COLOR = { r: 0.8, g: 0.8, b: 0, a: 1 };
+
+const styles = () => ({
+  container: {
+    height: '800px',
+    width: '1240px',
+    display: 'flex',
+    alignItems: 'stretch',
+  },
+});
+
+class AuditoryCortexExample extends Component {
+  constructor (props) {
+    super(props);
+    this.canvasRef = React.createRef();
+    this.state = {
+      hasModelLoaded: false,
+      showLoader: true,
+      data: [
+        {
+          instancePath: 'acnet2.baskets_12',
+          color: COLORS[1],
+        },
+        { instancePath: 'acnet2' },
+        {
+          instancePath: 'acnet2.baskets_12[0]',
+          color: COLORS[2],
+        },
+        {
+          instancePath: 'acnet2.baskets_12[7]',
+          color: COLORS[3],
+        },
+      ],
+      selected: {},
+      cameraOptions: {
+        angle: 60,
+        near: 10,
+        far: 2000000,
+        baseZoom: 1,
+        cameraControls: {
+          instance: CameraControls,
+          props: { wireframeButtonEnabled: false, },
+        },
+        reset: false,
+        autorotate: false,
+        wireframe: false,
+        position: { x: 230.357, y: 256.435, z: 934.238 },
+        rotation: { rx: -0.294, ry: -0.117, rz: -0.02, radius: 531.19 },
+      },
+    };
+
+    this.lastCameraUpdate = null;
+    this.cameraHandler = this.cameraHandler.bind(this);
+    this.selectionHandler = this.selectionHandler.bind(this);
+    this.hoverHandler = this.hoverHandler.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+
+  }
+
+  componentDidMount () {
+    import(/* webpackChunkName: "acnet_model.json" */'./acnet_model.json').then(model => {
+      GEPPETTO.Manager.loadModel(model);
+      Instances.getInstance(INSTANCE_NAME);
+      this.setState({ hasModelLoaded: true, showLoader: false })
+      document.addEventListener('mousedown', this.handleClickOutside);
+    })
+  }
+  
+  componentWillUnmount () {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+  
+  handleClickOutside (event) {
+
+    if (this.node && !this.node.contains(event.target)) {
+      if (event.offsetX <= event.target.clientWidth){
+        this.setState({ hasModelLoaded: false })
+      }
+    }
+  }
+
+  handleToggle () {
+    this.setState({ showLoader: true })
+
+    import(/* webpackChunkName: "acnet_model.json" */'./acnet_model.json').then(model => {
+      GEPPETTO.Manager.loadModel(model);
+      Instances.getInstance(INSTANCE_NAME);
+      this.setState({ hasModelLoaded: true, showLoader: false })
+    })
+  }
+
+  cameraHandler (obj) {
+    this.lastCameraUpdate = obj;
+  }
+
+  selectionHandler (selectedMap) {
```

Comment:
we should generalize this at some point and provide a default behavior and think about how to simplify custom handlers

cannot expect devs to have such in depth knowledge about the Canvas :D

Reply:
👍